gulpfile.js

```diff
@@ -0,0 +1,140 @@
+const exec = require('child_process').exec;
+const fs = require('fs');
+const readline = require('readline');
+
+const cwsupload = require('chrome-webstore-upload');
+const eslint = require('gulp-eslint');
+const {google} = require('googleapis');
+const gulp = require('gulp');
+const jeditor = require('gulp-json-editor');
+const zip = require('gulp-zip');
+
+const CWS_ID = 'aojcceglbipehndciapjedoomockgagl';
+
+/**
+ * Checks the code for style errors.
+ */
+gulp.task('lint', () => {
+  return gulp.src(['src/**/*.js', 'gulpfile.js', '!node_modules/**'])
+      .pipe(eslint({
+        baseConfig: require('./.eslintrc.js')
+      }))
+      .pipe(eslint.format())
+      .pipe(eslint.failAfterError());
+});
+
+/**
+ * Updates the Chrome Web Store manifest to point to the latest version of the
+ * scriipt.
+ */
+gulp.task('latest', () => {
+  // Determine the latest version of the script.
+  exec('clasp versions', (err, out) => {
+    if (err) return console.error(err);
+    const line = out.trim().split('\n').pop();
+    const version = line.split(' ').shift();
+
+    // Update the manifest.
+    gulp.src('webstore/manifest.json')
+      .pipe(jeditor((json) => {
+        json.container_info.container_version = version;
+        return json;
+      }))
+      .pipe(gulp.dest('webstore'));
+  });
+});
+
+/**
+ * Bumps the version of the Chrome Web Store manifest.
+ */
+gulp.task('bump', (cb) => {
+  gulp.src('webstore/manifest.json')
+    .pipe(jeditor((json) => {
+      json.version = String(Number(json.version) + 1);
+      return json;
+    }))
+    .pipe(gulp.dest('webstore'))
+    .on('end', cb);
+});
+
+/**
+ * Builds a new CRX zip file.
+ */
+gulp.task('compress', ['bump'], (cb) => {
+  gulp.src('webstore/*')
+    .pipe(zip('crx.zip'))
+    .pipe(gulp.dest('build')
+    .on('end', cb));
+});
+
+/**
+ * Uploads the CRX zip file to the Chrome Web Store
+ */
+gulp.task('upload', ['compress'], (cb) => {
+  var webStore = getWebStore();
+  const crx = fs.createReadStream('./build/crx.zip');
+  webStore.uploadExisting(crx).then((res) => {
+    console.log('Draft uploaded to Chrome Web Store.');
+    cb();
+  }).catch((e) => {
```

Comment:
This statement is unwrapping `e` then wrapping it back in the `error` function.

You can just pass the `error` function rather than creating a wrapper function:
```js
}).catch(console.error);
```

Reply:
Realized I should have passed the error to the callback, so that tasks that depend on this don't continue.